### ISSUE TITLE
poor rendering performance with compton

### ISSUE THREAD

**Issue by antis81** [lxqt/qterminal#293]
System info:

Component | Version
---- | ----
Hardware | Macbook 2,1 (late 2007, plastic) ; Graphics-Chipset: Intel 945GM (OpenGL 1.4, Mesa 11.2)
Distro | Lubuntu 16.04 (Linux kernel 4.4.0-53)
LXQt | 0.10.0 (from official Ubuntu repo)

Description:
I am using QTerminal as primary terminal emulator for ~2 years now and in general like it very much and had basically no problems with it. There's only one thing, that is very annoying and keeps appearing throughout LXQt versions 0.8.0 to 0.10.0. While every application (Qt or Gtk) performs nicely in the means of graphics rendering, QTerminal always had a very poor rendering performance, which appears when dragging or resizing the terminal window - even when dragging a focused window over it.

In short:
Disabling Compton improves performance, which I use only with the rather simple drop-shadow effect - no transparency, blurring or anything else. As mentioned above, it works smooth and without major performance hits with in general every application, but only causes trouble in QTerminal.

Apparently nobody else seems to have reported on this issue. Sorry, I don't have the latest and greatest gaming machine available… :innocent:

If that helps, I would happily compile and test upstream code. Would need some "getting started" help in this case (it's been a while since my last contrib's…).

**Comment by yan12125** [comment]
I guess the problem is not in QTerminal as it's a rather simple application. Does the performance issue appear on other Qt5 applications?

**Comment by antis81** [comment]
As I wrote in the description, the behaviour just appears with QTerminal. Other Qt5 applications don't show any visible issues. The issue is definitely related to QTerminal. After some more investigation it seems to come down to QTermWidget, as QTerminal's dialogs behave "normal".

**Comment by yan12125** [comment]
If you want to test qtermwidget only, clone https://github.com/lxde/qtermwidget and build with ```cmake -DBUILD_TEST .``` For newer cmake versions, you may want to change ```test``` info ```qtermwidget_test``` if there are errors:
```Diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6549739..3f2009d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,10 +179,10 @@ install(FILES
 # test application
 if(BUILD_TEST)
     set(TEST_SRC src/main.cpp)
-    add_executable(test ${TEST_SRC})
-    add_dependencies(test ${QTERMWIDGET_LIBRARY_NAME})
+    add_executable(qtermwidget_test ${TEST_SRC})
+    add_dependencies(qtermwidget_test ${QTERMWIDGET_LIBRARY_NAME})
     link_directories(${CMAKE_BINARY_DIR})
-    target_link_libraries(test ${QTERMWIDGET_QT_LIBRARIES} ${QTERMWIDGET_LIBRARY_NAME} util)
+    target_link_libraries(qtermwidget_test ${QTERMWIDGET_QT_LIBRARIES} ${QTERMWIDGET_LIBRARY_NAME} util)
 endif (BUILD_TEST)
 # end of test application
```

Does the built program suffers, too?
And what's your Qt version? Here on Arch with Qt 5.7, everything looks smooth.

**Comment by antis81** [comment]
>Does the built program suffers, too?

Will check that, if it hopefully still builds (it's been more than one year ago).

>And what's your Qt version? Here on Arch with Qt 5.7, everything looks smooth.

Yep -> Qt 5.7 (have some doubts on the "smoothness" :snail:, but let's see how it goes…).

**Comment by antis81** [comment]
@yan12125 Interesting… I fetched recent source for libqtermwidget and built the test application on current master (commit ) as you suggested. Result: Performance appears pretty ok!

Unfortunately I couldn't build QTerminal out-of-box right now. It is missing `lxqt-build-tools` and `liblxqt-dev` and probably that kde5 lib (what's the name again?). Could you give advice on what is needed to link, where to place the libs and which paths to setup?

**Comment by yan12125** [comment]
Yes you need to install lxqt-build-tools first from https://github.com/lxde/lxqt-build-tools/ with standard cmake build steps. liblxqt-dev is no longer necessary in the latest versoin. If there are still errors, paste the build log.

**Comment by agaida** [comment]
antis - try another compton rendering backend - if not done yet - one can set this in the compton configuration (the text file)

**Comment by antis81** [comment]
@agaida Hm… thanks, but it isn't very helpful. However I changed the backend giving "glx" another chance and finally it doesn't render the desktop unusable at least, which is definitely an improvement on that end - yay! :smile:

FYI: There's no special compton-rule or anything configured for QTerminal and it is rendered just like any other application window. The "brake" definitely sits in QTerminal (overridden paintEvent??). It becomes worse in turn, when opening multiple QTerminal instances. So any workaround is just a question of how many instances it needs to prove it wrong (by 3 my system already heavily struggles). This does not happen with e.g. pcmanfm-qt or any other application. I cannot say 100% sure, but it seems to be ok with the "frameless terminal widget" - IOW: It points to something weird going on in QMainWindow::paintEvent…

**Comment by agaida** [comment]
@antis81 - if there is no special reason for openbox one could try a different WM - compiz, xfwm4 or kwin are good choices. And depending on your distribution it might be a good idea to build the latest compton yourself.
